packages/verified-fetch/src/utils/parse-url-string.ts

```diff
@@ -89,30 +97,43 @@ export async function parseUrlString ({ urlString, ipns, logger }: ParseUrlStrin
   let cid: CID | undefined
   let resolvedPath: string | undefined
   const errors: Error[] = []
+  let resolveResult: IPNSResolveResult | DNSLinkResolveResult | undefined
+  let ttl: number = 5 * 60 // 5 minutes based on https://github.com/ipfs/boxo/issues/329#issuecomment-1995236409
 
   if (protocol === 'ipfs') {
     try {
       cid = CID.parse(cidOrPeerIdOrDnsLink)
+      ttl = 29030400 // 1 year for ipfs content
     } catch (err) {
       log.error(err)
       errors.push(new TypeError('Invalid CID for ipfs://<cid> URL'))
     }
   } else {
-    let resolveResult = ipnsCache.get(cidOrPeerIdOrDnsLink)
+    // protocol is ipns
+    resolveResult = ipnsCache.get(cidOrPeerIdOrDnsLink)
 
     if (resolveResult != null) {
       cid = resolveResult.cid
       resolvedPath = resolveResult.path
+      const answerTtl = (resolveResult as DNSLinkResolveResult).answer?.TTL
+      const recordTtl = (resolveResult as IPNSResolveResult).record?.ttl
+      ttl = Number(answerTtl ?? recordTtl ?? ttl)
       log.trace('resolved %s to %c from cache', cidOrPeerIdOrDnsLink, cid)
     } else {
-      // protocol is ipns
       log.trace('Attempting to resolve PeerId for %s', cidOrPeerIdOrDnsLink)
       let peerId = null
       try {
         peerId = peerIdFromString(cidOrPeerIdOrDnsLink)
         resolveResult = await ipns.resolve(peerId, { onProgress: options?.onProgress })
         cid = resolveResult?.cid
         resolvedPath = resolveResult?.path
+        /**
+         * For TTL nuances, see
+         *
+         * @see https://github.com/ipfs/js-ipns/blob/16e0e10682fa9a663e0bb493a44d3e99a5200944/src/index.ts#L200
+         * @see https://github.com/ipfs/js-ipns/pull/308
+         */
+        ttl = Number(resolveResult.record.ttl)
         log.trace('resolved %s to %c', cidOrPeerIdOrDnsLink, cid)
         ipnsCache.set(cidOrPeerIdOrDnsLink, resolveResult, 60 * 1000 * 2)
```

Comment:
We should use the record's TTL value also for the TLRU cache (instead of the static 2 minute value we set).

Reply:
see https://github.com/ipfs/helia-verified-fetch/pull/23/files#r1526281785 for discussion